### PR TITLE
Update To Angular 10

### LINKED ISSUE #24: JDNConvertibleCalendarDateAdapter not compatible with angular 10
I got this error while tying to update my angular 9 project to angular 10. 

 Package "jdnconvertiblecalendardateadapter" has an incompatible peer dependency to "@angular/material" (requires "^9.2.1", would install "10.2.0").

### PR DESCRIPTION
**Pull request by tobiasschweizer**
resolves #24

### CODE REVIEW

**Comment by tobiasschweizer** [comment]
@Gayashani95 This PR updates the lib from Angular 9 to 10. Could you maybe try to use it in your app?

You can do so using [`yalc`](https://www.npmjs.com/package/yalc):
- from this repo's root dir, run `npm run yalc-publish`
- then in your app, run `yalc add jdnconvertiblecalendardateadapter` and `npm i`

I will have to fix this issue in jdnconvertiblecalendar by changing `tsconfig.json`: 
> WARNING in .../JDNConvertibleCalendarDateAdapter/src/app/app.component.ts depends on 'jdnconvertiblecalendar'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

**Comment by tobiasschweizer** [comment]
@Gayashani95 Did you have a chance to try it out?

**Comment by Lexa950609** [comment]
Hi,
Yes I tried several times but the issue is still there. It gives me this
console error when I run the application..
I had to disable the hijri calendar temporarily due to this problem. My
company has been using  those two libraries for a long time (
JDNConvertibleCalendarDateAdapter and JDNConvertibleCalendar ) And my
techlead told me to find a solution quickly.. So,, it would be very helpful
for me if you give a publish for those libraries asap.

Thanks
Best Regards

**Comment by tobiasschweizer** [comment]
> Hi, Yes I tried several times but the issue is still there. It gives me this console error when I run the application.. I had to disable the hijri calendar temporarily due to this problem. My company has been using those two libraries for a long time ( JDNConvertibleCalendarDateAdapter and JDNConvertibleCalendar ) And my techlead told me to find a solution quickly.. So,, it would be very helpful for me if you give a publish for those libraries asap. Thanks Best Regards
> […](#)

So when you try to locally build and publish from this branch, you are still getting the issue?

**Comment by tobiasschweizer** [comment]
@Gayashani95 This branch will give you a warning because the current version of `JDNConvertibleAdapter` is built with "commonjs", but it should work without any problems.

Ok, I will publish a new version for `JDNConvertibleDateAdapter` still using the current version of `JDNConvertibleCalendar` once this PR is merged.

File: package.json [hunk 1]
```diff
@@ -15,43 +15,43 @@
...
-    "@angular/language-service": "^9.1.2",
+    "@angular-devkit/build-angular": "~0.1001.0",
+    "@angular-devkit/build-ng-packagr": "~0.1001.0",
+    "@angular/cli": "^10.1.0",
+    "@angular/compiler-cli": "^10.1.0",
+    "@angular/language-service": "^10.1.0",
     "@types/jasmine": "~2.8.6",
     "@types/jasminewd2": "~2.0.3",
     "@types/node": "^12.11.1",
     "codelyzer": "^5.1.2",
```
Comment (flavens) [inline package.json:43]: I get the following warning during the install: `npm WARN codelyzer@5.2.2 requires a peer of @angular/compiler@>=2.3.1 <10.0.0 || >9.0.0-beta <10.0.0 || >9.1.0-beta <10.0.0 || >9.2.0-beta <10.0.0 but none is installed. You must install peer dependencies yourself.` 
Version **6.0.0** will fix it.

**Review by flavens — CHANGES_REQUESTED** [review]
LGTM, one fix to do:

File: package.json (hunk 1 above)
Comment (tobiasschweizer) [reply package.json:43]: Hm, works for me.

Could you try removing `node_modules` first and then run `npm i`?

Comment (tobiasschweizer) [reply package.json:43]: this is the log from travis:

https://github.com/dhlab-basel/JDNConvertibleCalendarDateAdapter/runs/1149505780

Comment (flavens) [reply package.json:43]: I do not have the error when I delete the node mdules, but as soon as I re-run `npm i` I get the error. If this is ok for you, request my review again

Comment (tobiasschweizer) [reply package.json:43]: Ok, same for me now.

However, I also get these warnings in dsp-ui when I re-run `npm i`. Honestly: no idea why.

File: projects/jdnconvertible-calendar-date-adapter/package.json [hunk 2]
```diff
@@ -18,12 +18,15 @@
     "url": "https://github.com/dhlab-basel/JDNConvertibleCalendarDateAdapter/issues"
   },
   "homepage": "https://github.com/dhlab-basel/JDNConvertibleCalendarDateAdapter#readme",
+  "dependencies": {
+    "tslib": "^2.0.0"
```
Comment (tobiasschweizer) [inline projects/jdnconvertible-calendar-date-adapter/package.json:22]: I wondered about this direct dep: https://angular.io/guide/migration-update-libraries-tslib

File: package.json (hunk 1 above)
Comment (flavens) [reply package.json:43]: I do not get this error anymore when I use `"codelyzer": "^6.0.0"` but there are other warnings and I cannot be sure where they come from

Comment (tobiasschweizer) [reply package.json:43]: these are dev deps, right? So they shouldn't affect production

Comment (flavens) [reply package.json:43]: no they should not

Comment (tobiasschweizer) [reply package.json:43]: I checked, and we have the same version of codelyzer in dsp-ui. It was not updated during the Angular migration.

Comment (tobiasschweizer) [reply package.json:43]: ok, I create a DSP task so we can observe this when doing the Angular 10 update in dsp-ui: https://dasch.myjetbrains.com/youtrack/issue/DSP-691

**Review by flavens — APPROVED** [review]

**Comment by praveenInvenio-dev** [comment]
Hi @tobiasschweizer the first month in islamic calendar should have 29 days but in this calendar we are getting 30 days. Can you please help me override the months dates

**Comment by tobiasschweizer** [comment]
> Hi @tobiasschweizer the first month in islamic calendar should have 29 days but in this calendar we are getting 30 days. Can you please help me override the months dates

Hi, could you please open a new issue in this repo with some infos about your setup and a screenshot?

**Pull request merged**